src/main/java/org/javarosa/core/model/utils/DateUtils.java

```diff
@@ -221,6 +221,10 @@ private static Date getDate(DateFields df, int timezoneOffset) {
         cd.set(Calendar.SECOND, df.second);
         cd.set(Calendar.MILLISECOND, df.secTicks);
 
+        if (df.hour == 0 && df.minute == 0 && df.second == 0 && df.secTicks == 0) {
```

Comment:
This feels like it can't always be the right thing to do. For instnace, what if the time happens to actually have been meant to be midnight?